text/_util.ts

```diff
@@ -0,0 +1,129 @@
+ * @returns A sorted copy of possibleWords
+ * @note
+ * the ordering of words may change with version-updates
+ * e.g. word-distance metric may change (improve)
+ * use a named-distance (e.g. levenshteinDistance) to
+ * guarantee a particular ordering
+ */
+export function closest(
+  givenWord: string,
+  possibleWords: string[],
+  options?: {
+    caseSensitive?: boolean;
+  },
+): string {
+  assert(
+    possibleWords.length > 0,
+    `When using closest(), the possibleWords array must contain at least one word`,
+  );
+  const { caseSensitive } = { ...options };
+
+  if (!caseSensitive) {
+    givenWord = givenWord.toLowerCase();
+  }
+
+  let nearestWord = possibleWords[0];
+  let closestDistance = 0;
+  for (const each of possibleWords) {
+    const distance = caseSensitive
+      ? getWordDistance(givenWord, each)
+      : getWordDistance(givenWord, each.toLowerCase());
+    if (distance < closestDistance) {
+      nearestWord = each;
+      closestDistance = distance;
+    }
+  }
+  // this distance metric could be swapped/improved in the future
+  return nearestWord;
+}
+
+/**
+ * Sort based on word similarity
+ *
+ * @example
+ * ```ts
+ * import { similarityCompare } from "https://deno.land/std@$STD_VERSION/text/_util.ts";
+ * const words = ["hi", "hello", "help"];
+ *
+ * // words most-similar to "hep" will be at the front
+ * words.sort(similarityCompare("hep"));
+ * ```
+ * @note
+ * the ordering of words may change with version-updates
+ * e.g. word-distance metric may change (improve)
+ * use a named-distance (e.g. levenshteinDistance) to
+ * guarantee a particular ordering
+ */
+export function similarityCompare(
+  givenWord: string,
+  options?: { caseSensitive?: boolean },
+) {
+  const { caseSensitive } = { ...options };
+  if (caseSensitive) {
+    return (a: string, b: string) =>
+      getWordDistance(givenWord, a) - getWordDistance(givenWord, b);
+  }
+  givenWord = givenWord.toLowerCase();
+  return (a: string, b: string) =>
+    getWordDistance(givenWord, a.toLowerCase()) -
+    getWordDistance(givenWord, b.toLowerCase());
+}
+
+/**
+ * Sorts a string-array by similarity to a given string
+ *
+ * @example
+ * ```ts
+ * import { wordSimilaritySort } from "https://deno.land/std@$STD_VERSION/text/_util.ts";
+ *
+ * const possibleWords = ["length", "size", "blah", "help"];
+ *
+ * // case-insensitive by default
+ * const suggestions = wordSimilaritySort("hep", possibleWords).join(", ");
+ *
+ * // force case sensitive
+ * wordSimilaritySort("hep", possibleWords, { caseSensitive: true });
+ * ```
+ *
+ * @param givenWord - The string to measure distance against
+ * @param possibleWords - The string-array that will be sorted
+ * @param options.caseSensitive - Flag indicating whether the distance should include case. Default is false.
+ * @returns {string[]} A sorted copy of possibleWords
+ */
+export function wordSimilaritySort(
```

Comment:
I'll leave that up to you guys, its the difference between:

```
let suggestions = [...possibleWords].sort(similarityCompare("hep", { caseSensitive })).join(", ")
```
and 
```
let suggestions = wordSimilaritySort("hep", possibleWords).join(", ")
```

Reply:
I'm not understanding why one would need to do `[...possibleWords]`?

Reply:
cause otherwise it mutates `possibleWords`, which might be sorted in a particular order already (for example an order that is used when the help command is invoked)

Reply:
Ah sorry, I should have clarified. Why not use `.toSorted()`? It would just be
```ts
let suggestions = possibleWords.toSorted(similarityCompare("hep", { caseSensitive })).join(", ")
```
which is a little more readable imo.

Reply:
I'd prefer older compatibility (e.g. avoiding bleeding-edge ES2023 features), and plus toSorted is literally only one character shorter than the more explicit version.